samples/bluetooth/mesh/common/smp_bt_auth.c

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/init.h>
+
+/* .. include_startingpoint_mesh_smp_dfu_rst_1 */
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/conn.h>
+#include <bluetooth/mesh/vnd/le_pair_resp.h>
+
+static void passkey_display(struct bt_conn *conn, unsigned int passkey)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	printk("Please enter passkey (%d) on %s\n", passkey, addr);
+}
+
+static void cancel(struct bt_conn *conn)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	printk("Pairing cancelled: %s\n", addr);
+}
+
+static void pairing_confirm(struct bt_conn *conn)
+{
+	int err;
+
+	/* Automatically confirm pairing request from the device side. */
+	err = bt_conn_auth_pairing_confirm(conn);
+	if (err) {
+		printf("Can't confirm pairing (err: %d)\n", err);
+		return;
+	}
+
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	printk("Pairing confirmed: %s\n", addr);
+}
+
+static struct bt_conn_auth_cb auth_cb = {
+	/* Enable passkey_display callback to enable the display capability. */
```

Comment:
This is just for debugging purposes, so it will be nice to state that to avoid misinterpretation. You can also mention this instead at the 'printk' which actually prints it.
```suggestion
	/* Enable passkey_display callback to enable the display capability for debugging or developmental purposes. */
```

Reply:
No, this is not. This enables display capability: https://github.com/zephyrproject-rtos/zephyr/blob/3d3b287b7908625dc8356ac3532d9b6943edbb50/include/zephyr/bluetooth/conn.h#L1393-L1397